common/utils/src/main/scala/org/apache/spark/util/SparkFileUtils.scala

```diff
@@ -134,6 +134,16 @@ private[spark] trait SparkFileUtils extends Logging {
     createDirectory(root, namePrefix)
   }
 
+  def createParentDirs(file: File): Unit = {
+    if (file == null) {
+      throw new IllegalArgumentException("Input should not be null.")
+    }
+    val parent = file.getParentFile()
+    if (parent != null) {
+      Files.createDirectories(parent.toPath())
```

Comment:
What's the difference between this and `parent.mkdirs()`?

Btw, should we handle the following like `com.google.common.io.Files`?

```

    if (!parent.isDirectory()) {
      throw new IOException("Unable to create parent directories of " + file);
    }
```

Reply:
`Files.createDirectories` is a newer API than `File.mkdirs`.  In our case, we just want to make it sure that the directory exists. Since `Files.createDirectories` doesn't complain on the existing directory. That works for our use case.